src/dpo/src/detailed_manager.cxx

```diff
@@ -1384,6 +1384,84 @@ int DetailedMgr::checkEdgeSpacingInSegments()
   return err_n + err_p;
 }
 
+std::vector<std::vector<int>> DetailedMgr::getOneSiteGapViolationsPerSegment()
+{
+  // the pair is the segment and the cell index in that segment
+  std::vector<std::vector<int>> violating_cells;
+  violating_cells.resize(segments_.size() + 1);
+
+  std::vector<Node*> temp;
+  temp.reserve(network_->getNumNodes());
+  for (int s = 0; s < segments_.size(); s++) {
+    // To be safe, gather cells in each segment and re-sort them.
+    temp.clear();
+    for (Node* ndj : cellsInSeg_[s]) {
+      temp.push_back(ndj);
+    }
+    std::sort(temp.begin(), temp.end(), compareNodesX());
```

Comment:
compareNodesX is not a total order so it would be better to use stable_sort or to make it into one (ie handle tie breaking)

Reply:
This one is still open

Reply:
resolved